src/phoenix/server/api/mutations/dataset_mutations.py

```diff
@@ -300,6 +305,112 @@ async def delete_dataset(
             )
         )
 
+    @strawberry.mutation
+    async def patch_dataset_examples(
+        self,
+        info: Info[Context, None],
+        input: PatchDatasetExamplesInput,
+    ) -> DatasetMutationPayload:
+        if not (patches := input.patches):
+            raise ValueError("Must provide examples to patch.")
+        example_ids = [
+            from_global_id_with_expected_type(patch.example_id, DatasetExample.__name__)
+            for patch in patches
+        ]
+        if len(set(example_ids)) < len(example_ids):
+            raise ValueError("Cannot patch the same example more than once per mutation.")
+        if any(patch.is_empty() for patch in patches):
+            raise ValueError("Received one or more empty patches that contain no fields to update.")
+        version_description = input.version_description or None
+        version_metadata = input.version_metadata or {}
+        async with info.context.db() as session:
+            datasets = (
+                await session.scalars(
+                    select(models.Dataset)
+                    .join(
+                        models.DatasetExample,
+                        onclause=models.Dataset.id == models.DatasetExample.dataset_id,
+                    )
+                    .where(
+                        models.DatasetExample.id.in_(example_ids),
+                    )
+                )
```

Comment:
nit.
```suggestion
                await session.scalars(
                    select(models.Dataset)
                    .where(
                        models.Dataset.id.in_(
                            select(distinct(models.DatasetExample.dataset_id))
                            .where(models.DatasetExample.id.in_(example_ids))
                            .scalar_subquery()
                        )
                    )
                    .limit(2)
                )
```
Alternatively we can just combine this query with the next one, since most case are happy paths in which duplication is not an issue

Reply:
Changed to use `distinct` in d8e0aa6837f71d74b7e4f76ae49844d464f2dd9c. Not sure I see how to combine with the query that follows without sending extra data over the wire.

Reply:
> sending extra data over the wire

I was referring to the fact that most of the time that is ok because it would be the happy case and extra data is not sent. Extra data is sent in the edge case but that doesn't happen very often.

Reply:
It would still be extra in the happy path case since we would be sending an entire dataset object over the wire for each revision, no? Unless `sqlalchemy` is doing some caching.